src/test.js

```diff
@@ -218,7 +218,8 @@ Test.prototype = {
 			// The 'after' hook should only execute when there are not tests left and
 			// when the 'after' and 'finish' tasks are the only tasks left to process
 			if ( hookName === "after" &&
-				hookOwner.unskippedTestsRun !== numberOfUnskippedTests( hookOwner ) - 1 &&
+				( hookOwner.unskippedTestsRun + hookOwner.testsNotRun ) !==
+					numberOfUnskippedTests( hookOwner ) - 1 &&
```

Comment:
Should we still have this condition? Why wouldn't skipped tests be treated the same as ones that were filtered out (and therefore be included in `testsNotRun` directly)?

Reply:
This check was still needed to make sure we only run the `after` hook ON the last test of the module. (hence the `- 1` from `numberOfUnskippedTests( hookOwner ) - 1`

I have added an equivalent check with
```
function lastTestWithinModuleExecuted( module ) {
	return module.testsRun + module.testsIgnored === collectTests( module ).length - 1;
}
```